facilitator/src/aggregation.rs

```diff
@@ -123,9 +134,17 @@ impl<'a> BatchAggregator<'a> {
             .map(|k| Server::new(ingestion_header.bins as usize, self.is_first, k.clone()))
             .collect::<Result<_, _>>()?;
 
-        for batch_id in batch_ids {
-            self.aggregate_share(&batch_id.0, &batch_id.1, &mut servers, &mut invalid_uuids)?;
-            included_batch_uuids.push(batch_id.0);
+        let mut ingestion_batch = Some((ingestion_header.clone(), ingestion_packets));
+        for (batch_id, batch_date) in batch_ids_and_dates {
+            self.aggregate_share(
+                batch_id,
+                batch_date,
+                &mut servers,
+                &mut invalid_uuids,
+                ingestion_batch,
+            )?;
+            ingestion_batch = None;
+            included_batch_uuids.push(batch_id.to_owned());
```

Comment:
Yep, that's better, though I had to do a little more refactoring than I thought I would have to to make the borrow checker happy. Now I do all the reads in `BatchAggregator::generate_sum_part`, passing only the packets to `BatchAggregator::aggregate_share`.

The one wrinkle is the creation of the servers. Creating the servers requires data from an ingestion header; so I have to fill in an `Option` if unfilled. Similarly, the `ingestion_header` is required to do some post-aggregation work, so I have to stash it in an `Option` as well.

Reply:
Note that there's potential to panic in `BatchAggregator::generate_sum_part` if `batch_ids_and_dates` is empty. Previously we would panic when we tried to retrieve the first element; now we will panic when we try to unwrap `ingestion_header`. I'm not sure what the right remediation is: we could just return early, but then we wouldn't write anything (might be OK?). But we can't actually generate a sum part because the `SumPart` message derives a lot of its information from the `ingestion_header`, which we don't have  in this case.

Thoughts?

Reply:
I don't think we should ever get an empty `batch_ids_and_dates` in normal operation: if no aggregatable batches exist in a window, then `workflow-manager` shouldn't enqueue a task at all. Therefore I think it's reasonable for `facilitator` to simply fail with an error if the list of batches is empty.

Reply:
Agreed, `batch_ids_and_dates` will never be empty in normal operation. Added an error-return for this case, which will effectively cause the relevant message to end up in the DLQ if this ever does occur somehow (which is what we want, I suppose).

Reply:
BTW, while this worked out, it required quite a bit of refactoring to appease the borrow checker. One change that made things easier was making `Transport` methods non-mut, which triggered cascading waves of clippy complaints about e.g. "this variable does not need to be mut". Apologies -- I didn't realize how big of a refactoring this would become until after I was done, at which point it's sort of hard to unwind & pull out into its own commit/PR. :-/